File: PCKART-FRONTEND/ADMIN/src/app/_model-dto/product/product.ts
import { Brand } from "../brand/brand"
import { Category } from "../category/category"

export class Product{
  id: number;
  name: string; 
  brand: Brand;
  price: number;
  quantity: number;
  discount: number;
  category: Category;
  color: string;
  description: string;
  specs: any;
  added_at: any;
  active: boolean;
}

File: PCKART-BACKEND/PCKART-PROJECT-ADMIN/src/main/java/com/ar/pckart/product/model/Product.java
package com.ar.pckart.product.model;

import java.util.HashMap;
import java.util.Map;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Product {
	
	private Long id;
	private String name;
	private Brand brand;
	private double price;
	private int quantity;
	private float discount;
	private Category category;
	private String color;
	private String description;
	private Map<String,String> specs = new HashMap<>();
}
